3main

```diff
@@ -1089,72 +946,78 @@ then
-  then
-    echo "Auto Start Miner Set to YES"
-    if [[ $WTM_AUTO_SWITCH == YES ]]
-    then
-      echo ""
-      echo "LAUNCHING:  WTM AUTO SWITCH "
-      echo ""
-      cat <<EOF >${NVOC}/WTM.json
-{
- "output_dir": "${NVOC}",
- "WTM_URL": "$WTM_AUTO_SWITCH_URL",
- "WTM_COINS": "$WTM_AUTO_SWITCH_COINS",
- "WTM_MIN_DIFFERENCE": "$WTM_MIN_DIFFERENCE",
- "currency":"$WTM_CURRENCY"
-}
-EOF
-      cat <<EOF >${NVOC}/WTM_1.json
-{
- "output_dir": "${NVOC}",
- "WTM_URL": "$WTM_AUTO_SWITCH_URL",
- "WTM_COINS": "$WTM_AUTO_SWITCH_COINS",
- "WTM_MIN_DIFFERENCE": "$WTM_LOWER_MIN_DIFFERENCE",
- "currency":"$WTM_CURRENCY"
-}
-EOF
-      if ! ps ax | grep -q '[8]wtm_auto_switch'
-      then
-        screen -c ${NVOC}/screenrc-wtm -dmSL wtm bash ${NVOC}/8wtm_auto_switch
-        if [[ $LOCALorREMOTE == LOCAL ]]; then
-          echo ""
-          echo "process in screen wtm, guake terminal Tab (f12)"
-          if [[ ! $(ps ax | grep "tail -f ${NVOC}/nvoc_logs/8_wtmautoswitchlog" | grep -v grep) ]];then
-            guake -n ${NVOC}/8wtm_auto_switch -r WTM_AUTO_SWITCH -e "tail -f  ${NVOC}/nvoc_logs/8_wtmautoswitchlog"
-          fi
-        elif [[ $LOCALorREMOTE == REMOTE ]]; then
-          echo "process in screen wtm, 'nvOC wtm-log'"
-          echo "    "
-        fi
-      fi
-    fi
-  fi
 
   while ps ax | grep SCREEN | grep -v cpuminer | grep -q miner ; do
     echo "" | tee -a ${NVOC}/nvoc_logs/screenlog.0
-    echo "$(date) - Miner is running, 3main restarting miner" | tee -a ${NVOC}/nvoc_logs/screenlog.0
+    echo "$(date) - Miner is running, 3main killing miner" | tee -a ${NVOC}/nvoc_logs/screenlog.0
     ps ax | grep SCREEN | grep -v cpuminer | grep miner | awk '"miner" {print $1}' | xargs kill -9
     sleep 2
   done
 
-  echo ""
-  echo "LAUNCHING:  MINER"
-  echo "    "
-  bash ${NVOC}/0miner
-  echo "    "
-  echo "    "$(ps ax | grep -i screen | grep miner | grep -o 'SCREEN.*') | tee -a ${NVOC}/nvoc_logs/screenlog.0
-  echo "    "
+  if [[ $ALGO != UNDEFINED && ( $AUTO_SWITCH == NO || $AUTO_SWITCH == WTM_SWITCHING ) ]]
+  then
+    echo ""
+    echo "LAUNCHING:  MINER"
+    echo "    "
+    bash ${NVOC}/0miner
+    echo "    "
+    echo "    "$(ps ax | grep -i screen | grep miner | grep -o 'SCREEN.*') | tee -a ${NVOC}/nvoc_logs/screenlog.0
+    echo "    "
+  
+    if [[ $LOCALorREMOTE == REMOTE ]]
+    then
+      echo "Miner output: ./nvOC miner-log"
+      echo "    "
+      echo "    "
+    elif [[ $LOCALorREMOTE == LOCAL ]]
+    then
+      echo "Miner output: ./nvOC miner-log"
+      echo "    "
+      echo "    "
+      tail -f -n 1 ${NVOC}/nvoc_logs/screenlog.0
+    fi
+  elif [[ ${AUTO_SWITCH:0:7} == SALFTER ]]
+  then
+    echo "LAUNCHING:  ${AUTO_SWITCH} (will launch miner)"
+    echo ""
+  
+    while true
+    do
+      ${AUTO_SWITCH_CMD}
+      echo ""
+      cat ${NVOC}/current-profit
+      echo ""
+      if [[ $LOCALorREMOTE == LOCAL ]]
+      then
+        guake -n 1 -r ${AUTO_SWITCH} -e "screen -r miner"
```

Comment:
I did not change this because I thought there were some reasons to behave differently with salfter switch. For example I'm not sure if all screen miner commands in switcher code have already the argument for output log.

Reply:
I dont see any reasons, in my last test, with every GPU problem (I have a faulty GPU in my test rig)  when watchdog restart 3main a new salfter tab opens, I changed mine and tested with no problems.

Reply:
Look at the last line of a salfter switch, you'll see it's not directing screenlog to the right place.

Reply:
Oh, forgot, I also changed mph_switch miner command:
`  subprocess.call(("screen -c /home/m1/screenrc-miner -dmSL miner "+coin).split(" "))`